src/odemis/driver/actuator.py

```diff
@@ -3550,3 +3550,160 @@ def stop(self, axes=None):
+                             set(self.axes) - speed_axes)
+
+        # TODO: correctly modify the reference, speed and axes range attributes
+
+    # def _updateAxesRange(self):
+    #     """
+    #     Calculate the axes range of position VA from the given range of dep's.
+    #     """
+    #     # Calculate rectangular border points on the given range
+    #     min_x, max_x = self.axes["x"].range
+    #     min_y, max_y = self.axes["y"].range
+    #     coordinates = [[min_x, min_y],
+    #                    [min_x, max_y],
+    #                    [max_x, min_y],
+    #                    [max_x, max_y]]
+
+    #     # Calculate range based on the conversion
+    #     new_coordinates = list(map(self._convertPosFromdep, coordinates))
+
+    #     # Update default axes range
+    #     new_range = numpy.array(new_coordinates, dtype=float)
+    #     self.axes["x"].range = (min(new_range[:, 0]), max(new_range[:, 0]))
+    #     self.axes["y"].range = (min(new_range[:, 1]), max(new_range[:, 1]))
+
+    def _updatePosition(self, pos_dep):
+        """
+        update the position VA when the dep's position is updated
+        """
+        # TODO: this should be posture converted to SEM posture
+        # pos_sem = self.pm.to_posture(pos_dep, SEM_IMAGING)
+        # pos = self.pm.to_sample_stage_from_stage_position(pos_sem)
+        # logging.warning(f"Converted position from {pos_dep} to {pos_sem}, Updating SampleStage position to {pos}")
+
+        pos = self.pm.to_sample_stage_from_stage_position(pos_dep)
+        # it's read-only, so we change it via _value
+        self.position._set_value(pos, force_write=True)
+
+        # update related MDs
+        affects = ["ccd", "e-beam"] # roles
+        for a in affects:
+            try:
+                comp = model.getComponent(role=a)
+                if comp:
+                    md_pos = pos.get("x", 0), pos.get("y", 0)
+                    comp.updateMetadata({model.MD_POS: md_pos})
+            except Exception as e:
+                logging.error("Failed to update %s with new position: %s", a, e)
+
+    def _updateSpeed(self, dep_speed):
+        """
+        update the speed VA based on the dependency's speed
+        """
+        # stage_speed = self.pm.to_sample_stage_from_stage_movement(dep_speed)
+        self.speed._set_value(dep_speed, force_write=True)
+
+    def _updateReferenced(self, dep_refd):
+        """
+        update the referenced VA
+        """
+        refd = {
+            ax: dep_refd[ad] for ax, ad in self.axes.items() if ad in dep_refd
+        }
+
+        self.referenced._set_value(refd, force_write=True)
+
+    @isasync
+    def moveRel(self, shift: Dict[str, float], **kwargs):
+        """
+        :param shift: The relative shift to be made
+        :param **kwargs: Mostly there to support "update" argument
+        """
+        # missing values are assumed to be zero
+        shift_stage = self.pm.from_sample_stage_to_stage_movement(shift)
```

Comment:
I wonder if a relative move is actually possible to compute in every cases (TFS, probably yes, but I'm more concerned with Tescan and the complex transformations). It might be easier to convert to an absolute position, (as .position + shift) and then do the same as moveAbs().

Reply:
I don't understand why computing an absolute movement would be possible, but relative wouldn't?